pangeo_forge_recipes/recipes/xarray_zarr.py

```diff
@@ -191,67 +192,45 @@ def region_and_conflicts_for_chunk(
     concat_dim: str,
     metadata_cache: Optional[MetadataTarget],
     subset_inputs: Optional[SubsetSpec],
-) -> Tuple[Dict[str, slice], Set[int]]:
+) -> Tuple[Dict[str, slice], Dict[str, Set[int]]]:
     # return a dict suitable to pass to xr.to_zarr(region=...)
     # specifies where in the overall array to put this chunk's data
     # also return the conflicts with other chunks
 
-    ninputs = file_pattern.dims[file_pattern.concat_dims[0]]
-    input_keys = inputs_for_chunk(chunk_key, inputs_per_chunk, ninputs)
-
     if nitems_per_input:
         input_sequence_lens = (nitems_per_input,) * file_pattern.dims[concat_dim]  # type: ignore
     else:
         assert metadata_cache is not None  # for mypy
         global_metadata = metadata_cache[_GLOBAL_METADATA_KEY]
         input_sequence_lens = global_metadata["input_sequence_lens"]
+    total_len = sum(input_sequence_lens)
 
+    input_chunk_grid = ChunkGrid({concat_dim: input_sequence_lens})
     if subset_inputs and concat_dim in subset_inputs:
-        # scenario I: there is a single input per chunk, possibly with subsetting
         assert (
             inputs_per_chunk == 1
         ), "Doesn't make sense to have multiple inputs per chunk plus subsetting"
-        subset_factor = subset_inputs[concat_dim]
-        input_sequence_lens = tuple(
-            chain(*(calc_subsets(input_len, subset_factor) for input_len in input_sequence_lens))
-        )
-        subset_idx = [
-            dim_idx.index
-            for dim_idx in chunk_key
-            if dim_idx.operation == CombineOp.SUBSET and dim_idx.name == concat_dim
-        ][0]
```

Comment:
Note how we no longer need this opaque logic in this function. It now lives in ChunkGrid.

Reply:
(and the new `chunk_position` function, which translates a ChunkKey to a chunk index)